### PR TITLE
Allow setting commits on a release

### PR DESCRIPTION
**Pull request by a-type**
This is a revisiting of #507 with some feedback incorporated. Exposes the ability to add commits to a release.

### CODE REVIEW

**Comment by a-type** [comment]
Can a maintainer help me out with the failing CI? Looks like it's not compiling Rust, which I didn't touch (and also have never worked with).

**Comment by jan-auer** [comment]
Assigned @kamilogorek.

@a-type Does this mean we can close #507?

**Comment by a-type** [comment]
Sure, it this PR is preferred. I didn't open #507, I'm just trying to get this feature in.

File: js/releases/index.js [hunk 1]
```diff
@@ -33,6 +33,45 @@ module.exports = {
...
+   * @memberof SentryReleases
+   */
+  setCommits(release, options) {
+    if (!options || !options.repo || (!options.auto && !options.commit)) {
+      throw new Error(
+        'options.repo, and either options.commit or options.auto must be specified'
+      );
+    }
+
+    let commitFlags = [];
```
Comment (kamilogorek) [inline js/releases/index.js]: `commitFlags` will always be filled with proceeding `if/else` clause, so no need to predefine it.

File: js/releases/index.js [hunk 2]
```diff
@@ -33,6 +33,45 @@ module.exports = {
...
+   * the current commit). Overrides other options.
+   * @param {string} options.commit The current (last) commit in the release.
+   * @param {string} options.previousCommit The commit before the beginning of this
+   * release (in other words, the last commit of the previous release). If omitted,
+   * this will default to the last commit of the previous release in Sentry. If there
+   * was no previous release, the last 10 commits will be used.
+   * @returns {Promise} A promise that resolves when the commits have been associated
+   * @memberof SentryReleases
+   */
+  setCommits(release, options) {
```
Comment (kamilogorek) [inline js/releases/index.js]: `options` can be defaulted to `{}` and we can then skip `!options` check.

**Comment by jan-auer** [comment]
Going to merge this now as we're running a release. 
Thanks for the contribution!

**Pull request merged**